widgets/src/other/PropertyEditor.js

```diff
@@ -98,306 +115,705 @@
+            var sharedPropsMainSections = [];
+            var sPropSections = [];
+            if (this._data.length > 0) {
+                sharedPropsMainSections.push(findSharedProperties(this._data));
+
+                for (var k1 in sharedPropsMainSections) {
+                    var sectionArr = [];
+                    for (var k2 in sharedPropsMainSections[k1]) {
+                        var widgetArr = [];
+                        sharedPropsMainSections[k1][k2].arr.forEach(function(n) {
+                            widgetArr.push(n.widget);
+                        });
+                        if (this._share_count_min <= widgetArr.length || widgetArr[0]._class.indexOf('PropertyEditor') !== -1) {
+                            sectionArr.push({
+                                rowType: 'shared',
+                                widgetArr: widgetArr,
+                                id: sharedPropsMainSections[k1][k2].id,
+                                description: sharedPropsMainSections[k1][k2].description,
+                                type: sharedPropsMainSections[k1][k2].type,
+                                set: sharedPropsMainSections[k1][k2].set
+                            });
+                            sharedPropsMainSections[k1][k2].arr.forEach(function(widgetNode) {
+                                sectionArr.push({
+                                    rowType: 'individual',
+                                    widgetArr: [widgetNode.widget],
+                                    id: sharedPropsMainSections[k1][k2].id,
+                                    description: sharedPropsMainSections[k1][k2].description,
+                                    type: sharedPropsMainSections[k1][k2].type,
+                                    set: sharedPropsMainSections[k1][k2].set
+                                });
+                            });
+                        }
+                    }
+                    sPropSections.push(sectionArr);
+                }
+            }
+        }
+
+        //Creating Table and THEAD
+        if (true) {
+            var table = element.selectAll("#" + this._id + " > table").data(this._data, function(d) {
+                return d._id;
+            });
+            table.enter().append("table")
+                    .each(function(widget) {
+                        var element = d3.select(this);
+                        var thead = element.append("thead");
+                        thead.attr("class", "mm-label max").on("click", function() {
+                            var elm = this;
+                            if (hasClass(elm, 'min')) {
+                                addClass(elm, 'max');
+                                removeClass(elm, 'min');
+                            } else if (hasClass(elm, 'max')) {
+                                addClass(elm, 'min');
+                                removeClass(elm, 'max');
+                            }
```

Comment:
This could be:
  var elm = d3.select(this);
  if (elm.classed("min")) {
    elm.classed("max", true);
    elm.classed("min", false);
}...